### PR TITLE
Add a filter argument to grid functions.

### PR DESCRIPTION
**Pull request by adanvr**
Fix Issue #18

### CODE REVIEW

File: R/grids.R [hunk 1]
```diff
@@ -17,6 +17,8 @@
 #' @param original A logical: should the parameters be in the original units or
 #' in the transformed space (if any)?
 #'
+#' @param filter A logical: should the parameters be filtered prior to generating grid. Must be captured as a single condition.
```
[the remark this answers is not in the document]
Comment (DavisVaughan) [reply R/grids.R]: can we use "Must be a single expression referencing parameter names that evaluates to a logical vector"

File: R/grids.R [hunk 2]
```diff
@@ -32,7 +34,7 @@
 #' grid_random(penalty(), mixture())
 #'
```
Comment (DavisVaughan) [inline R/grids.R]: Can you please add an example of `filter` usage?

Comment (adanvr) [reply R/grids.R]: Yes, I have added an example and then ran devtools::document()

File: R/grids.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (DavisVaughan) [reply R/grids.R]: @adanvr did you see this comment?

[the remark this answers is not in the document]
Comment (DavisVaughan) [reply R/grids.R]: nevermind i got it

[the remark this answers is not in the document]
Comment (adanvr) [reply R/grids.R]: @DavisVaughan sorry I had to head out to airport so didn't see these but thanks so much!

Have a save flight home.

**Pull request merged**